src/lib.rs

```diff
@@ -315,6 +316,28 @@ pub trait SharedLibrary: Sized + Debug {
     /// Get the debug-id of this shared library if available.
     fn id(&self) -> Option<SharedLibraryId>;
 
+    /// Returns the address of where the library is loaded.
+    ///
+    /// This typically is the stated virtual memory address of the first
+    /// code segment.
+    fn stated_virtual_memory_address(&self) -> Svma {
+        self.segments()
+            .find(|x| x.is_code())
+            .map(|x| x.stated_virtual_memory_address())
+            .unwrap_or_else(|| Svma(ptr::null()))
+    }
+
+    /// Returns the size of the image.
+    ///
+    /// This typically is the size of the executable code segment.
+    fn len(&self) -> usize {
+        self.segments()
+            .skip_while(|x| !x.is_code())
+            .take_while(|x| x.is_code())
```

Comment:
I have similar concerns around clarity of intention for this method as well as the docs needing to mention discontiguous executable segments.

Reply:
The contiguous executable segments is again what systems like breakpad do. I am not sure why this happens in the first place but based on recent investigations this is likely to be the case when a part of the executable was loaded without executable bits. My plan is to follow this up with a more correct implementation.

Reply:
Also the point of knowing the image size is to know which instruction addresses definitely cannot be symbolicated by debug information associated with this image.